pages/proposals.tsx

```diff
@@ -0,0 +1,148 @@
+import { tzip16 } from "@taquito/tzip16";
+import { validateContractAddress } from "@taquito/utils";
+import { useContext, useEffect, useMemo, useState } from "react";
+import ProposalCard from "../components/ProposalCard";
+import Spinner from "../components/Spinner";
+import Meta from "../components/meta";
+import Modal from "../components/modal";
+import ProposalSignForm from "../components/proposalSignForm";
+import fetchVersion from "../context/metadata";
+import { getProposals } from "../context/proposals";
+import { AppStateContext } from "../context/state";
+import { proposal, version } from "../types/display";
+import { getProposalsId, toProposal } from "../versioned/apis";
+
+const emptyProps: [number, { og: any; ui: proposal }][] = [];
+
+const Proposals = () => {
+  const state = useContext(AppStateContext)!;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [invalid, setInvalid] = useState(false);
+  const [proposals, setProposals] = useState(emptyProps);
+  const [openModal, setCloseModal] = useState<{
+    state: number;
+    proposal: [boolean | undefined, number];
+  }>({
+    state: 0,
+    proposal: [undefined, 0],
+  });
+  const [refresher, setRefresher] = useState(0);
+
+  useEffect(() => {
+    if (!state.currentContract) return;
+
+    if (validateContractAddress(state.currentContract) !== 3) {
+      setInvalid(true);
+      return;
+    }
+
+    setIsLoading(true);
+
+    (async () => {
+      if (!state.currentContract) return;
+
+      let c = await state.connection.contract.at(state.currentContract, tzip16);
+
+      let cc = await c.storage();
+      let version = await (state.contracts[state.currentContract]
+        ? Promise.resolve<version>(
+            state.contracts[state.currentContract].version
+          )
+        : fetchVersion(c));
+
+      let bigmap: { key: string; value: any }[] = await getProposals(
+        getProposalsId(state.contracts[state.currentContract].version, cc)
+      );
+      let proposals: [number, any][] = bigmap.map(({ key, value }) => [
+        Number.parseInt(key),
+        { ui: toProposal(version, value), og: value },
+      ]);
+
+      setProposals(proposals);
+      setIsLoading(false);
+    })();
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.currentContract, refresher]);
+
+  const filteredProposals = useMemo(
+    () => [
+      ...proposals.filter(
+        ([_, proposal]) => "Proposing" === proposal.ui.status
+      ),
+    ],
+    [proposals]
+  );
+
+  const currentContract = state.currentContract ?? "";
+  return (
+    <div className="min-h-content relative flex grow flex-col">
```

Comment:
I think something wrong with CSS. The popped up Modal isn't in the center of screen. It's in the center of main part. It isn't easy to use when the high of screen is short.